src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -184,13 +185,20 @@ struct nonmaxsuppression
             result.y = {static_cast<double>(start[0]), static_cast<double>(start[2])};
         }
 
+        result.sort();
+
         return result;
     }
 
     inline bool suppress_by_iou(box b1, box b2, double iou_threshold) const
     {
-        b1.sort();
-        b2.sort();
+        const double area1 = b1.area();
+        const double area2 = b2.area();
+
+        if(area1 <= .0f or area2 <= .0f)
```

Comment:
Area can never be negative, given the current implementation of area(), so I am not sure what is this clause `<=` going to do here.  Perhaps, for an FP value, it is best to compare against an epsilon value, if the intent is to detect zeros.

Reply:
It is to check for zeros but in this cause should there be some error that double is less than zero erroneously, this would also return a false value.

Reply:
This reference should help point out the pitfalls of FP comparisons:
https://www.boost.org/doc/libs/1_78_0/libs/math/doc/html/math_toolkit/float_comparison.html
A FP check for zero could be written as:   
`if (area1 < std::numeric_limits<float>epsilon()) {  ... }`